### PR TITLE
Charlie Company Loadout Fixes

### LINKED ISSUE #425: Platoon Medic does not have appropriate medical permissions
Platoon Medic does not have proper medical permissions. More than likely an issue when we changed the loadouts.

### PR DESCRIPTION
**Pull request by Robot-Panda15**
**When merged this pull request will:**
- CHANGED: Charlie loadouts a major overhaul
- FIXED: Platoon medic permission not applied.
- CHANGED: Updated Charlie Arsenal Loadouts
**DON'T MERGE TESTING WILL BE PREFORMED BY COC**

**Snapshot release:**
- [cScripts-DEVBUILD-37cc7b2_TestBuild.zip](https://github.com/7Cav/cScripts/files/3665272/cScripts-DEVBUILD-37cc7b2_TestBuild.zip)

Fixes #425

### CODE REVIEW

**Comment by ColdEvul** [comment]
Fixed merge conflicts for you

**Comment by ColdEvul** [comment]
```
$ python3 tools/config_style_checker.py
Validating Config Style
ERROR: A possible missing curly brace { or } in file cScripts/Loadouts/CfgLoadouts_Charlie.hpp { = 137 } = 135
------
Checked 25 files
Errors detected: 1
Config validation FAILED
The command "python3 tools/config_style_checker.py" exited with 1.
```

File: cScripts/CavFnc/functions/selections/fn_initQuickSelections.sqf [hunk 1]
```diff
@@ -154,13 +154,11 @@ if (_companySelector in _charlie or _alwaysAvalible) then {
...
 
     [_object,"Fire Team Leader", "Cav_B_C_FireTeamLeader_F", "", _charlieSquadCat, "Charlie", _allowOnlyForCompany] call FUNC(addQuickSelection);
+    [_object,"Fire Team Leader M320", "Cav_B_C_FireTeamLeaderGrenadier_Local", "", _charlieSquadCat, "Charlie", _allowOnlyForCompany] call FUNC(addQuickSelection);
     [_object,"Automatic Rifleman", "Cav_B_C_AutomaticRifleman_F", "", _charlieSquadCat, "Charlie", _allowOnlyForCompany] call FUNC(addQuickSelection);
     [_object,"Grenadier", "Cav_B_C_Grenadier_F", "", _charlieSquadCat, "Charlie", _allowOnlyForCompany] call FUNC(addQuickSelection);
     [_object,"Rifleman (LAT)", "Cav_B_C_RiflemanLAT_F", "", _charlieSquadCat, "Charlie", _allowOnlyForCompany] call FUNC(addQuickSelection);
     [_object,"Designated Marksman", "Cav_B_C_Marksman_Local", "", _charlieSquadCat, "Charlie", _allowOnlyForCompany] call FUNC(addQuickSelection);
-
-    // Squad Extra
-    [_object,"Rifleman", "Cav_B_C_Rifleman_F", "", _charlieSquadCat, "Charlie", _allowOnlyForCompany] call FUNC(addQuickSelection);
```
Comment (ColdEvul) [inline cScripts/CavFnc/functions/selections/fn_initQuickSelections.sqf]: Re add this one even tho it's not a loadout we use it is sometimes usefull to have it. to base loadouts on or for sandbox or public reasons. 

The loadout can be hidden behind the mission type cba setting EF people realy whant ut gone.

File: cScripts/CavFnc/functions/selections/fn_initQuickSelections.sqf [hunk 2]
```diff
@@ -177,16 +175,18 @@ if (_companySelector in _charlie or _alwaysAvalible) then {
...
     [_object,"MMG FTL", "CAV_Charlie_Machinegunner_TeamLeader_Local", "", _charlieMMGTeamCat, "Charlie", _allowOnlyForCompany] call FUNC(addQuickSelection);
     [_object,"MMG Gunner", "CAV_Charlie_Machinegunner_Local", "", _charlieMMGTeamCat, "Charlie", _allowOnlyForCompany] call FUNC(addQuickSelection);
 
     // MAAWS Team
     [_object,"cScriptQuickSelection_Charlie_ATTeam","MAAWS Team",_icon,["ACE_MainActions","cScriptQuickSelectionMenu","cScriptQuickSelection_Charlie"]] call FUNC(addAceCategory);
     private _charlieATTeamCat = ['ACE_MainActions','cScriptQuickSelectionMenu','cScriptQuickSelection_Charlie','cScriptQuickSelection_Charlie_ATTeam'];
-    [_object,"MAAWS Gunner", "Cav_B_C_AT_Local", "", _charlieATTeamCat, "Charlie", _allowOnlyForCompany] call FUNC(addQuickSelection);
-    [_object,"MAAWS Assistant", "Cav_B_C_AAT_Local", "", _charlieATTeamCat, "Charlie", _allowOnlyForCompany] call FUNC(addQuickSelection);
+    [_object,"HE MAAWS Gunner", "Cav_B_C_HE_Local", "", _charlieATTeamCat, "Charlie", _allowOnlyForCompany] call FUNC(addQuickSelection);
```
Comment (ColdEvul) [inline cScripts/CavFnc/functions/selections/fn_initQuickSelections.sqf]: ATHE
AATAHE

File: cScripts/Loadouts/CfgLoadouts_Charlie.hpp [hunk 3]
```diff
@@ -367,20 +404,69 @@ class Cav_B_C_AAT_Local : Cav_B_Charlie_base_F { //MAAWS Assistant : 30kg
...
+    //gps[] = {""};
+    //map[] = {"ItemMap"};
+    nvgs[] = {"rhsusf_ANPVS_14"};
+    //watch[] = {"ItemWatch"};
+
+    //insignia[] = {""};
+    preLoadout = "[(_this select 0), 'charlie', 0, 0] call cScripts_fnc_setPreInitPlayerSettings;";
+    postLoadout = "[(_this select 0),true,true] call cScripts_fnc_setPostInitPlayerSettings;";
+};
+class Cav_B_C_AHE_Local : Cav_B_Charlie_base_F { //MAAWS HE Assistant : 36.22kg
```
Comment (ColdEvul) [inline cScripts/Loadouts/CfgLoadouts_Charlie.hpp]: AATHE

File: cScripts/Loadouts/CfgLoadouts_Charlie.hpp [hunk 4]
```diff
@@ -397,12 +483,12 @@ class Cav_B_C_AAT_Local : Cav_B_Charlie_base_F { //MAAWS Assistant : 30kg
     preLoadout = "[(_this select 0), 'charlie', 0, 0] call cScripts_fnc_setPreInitPlayerSettings;";
     postLoadout = "[(_this select 0),true,true] call cScripts_fnc_setPostInitPlayerSettings;";
 };
-class Cav_B_C_AT_Local : Cav_B_Charlie_base_F { //MAAWS Gunner : 37kg
+class Cav_B_C_AT_Local : Cav_B_Charlie_base_F { //MAAWS Gunner : 35.8kg
```
Comment (ColdEvul) [inline cScripts/Loadouts/CfgLoadouts_Charlie.hpp]: ATHE?

File: cScripts/Loadouts/CfgLoadouts_Charlie.hpp [hunk 5]
```diff
@@ -578,36 +718,40 @@ class Cav_B_C_Support_MK19_Assistant_Local : Cav_B_Charlie_base_F { //Mk.19 Tri
...
-        "rhs_mag_30Rnd_556x45_M855A1_Stanag_Tracer_Red",8,
+        "rhs_mag_30Rnd_556x45_M855A1_PMAG_Tracer_Red",7,
         "rhs_mag_m67",2,
         "SmokeShell",4,
-        "SmokeShellRed",2,
-        "SmokeShellGreen",2,
-        "rhsusf_100Rnd_762x51_m62_tracer",3
+        "SmokeShellRed",
+        "SmokeShellGreen",
+        "SmokeShellBlue,2,
```
Comment (ColdEvul) [inline cScripts/Loadouts/CfgLoadouts_Charlie.hpp]: Found you!

**Comment by ColdEvul** [comment]
Fixed #340

**Comment by Robot-Panda15** [comment]
Check Update Charlie Weapons Team Names commit for flaws

File: cScripts/CavFnc/functions/logistics/fn_doStarterCrateSupplies.sqf [hunk 6]
```diff
@@ -159,7 +159,7 @@ if (_companySelector in _alpha or _alwaysAvalible) then {
         //================== AMMO ==================\\
         _crate addMagazineCargoGlobal ["rhs_mag_30Rnd_556x45_M855A1_Stanag",250];
         _crate addMagazineCargoGlobal ["rhs_mag_30Rnd_556x45_M855A1_Stanag_Tracer_Red",50];
-        _crate addMagazineCargoGlobal ["rhs_200rnd_556x45_M_SAW",250];
+        _crate addMagazineCargoGlobal ["rhsusf_200rnd_556x45_mixed_box",250];
```
Comment (ColdEvul) [inline cScripts/CavFnc/functions/logistics/fn_doStarterCrateSupplies.sqf]: Self note: this needs to be done in the crate remake due to merge request later on.

File: cScripts/Loadouts/CfgLoadouts_Charlie.hpp [hunk 7]
```diff
@@ -167,80 +243,125 @@ class Cav_B_C_FireTeamLeader_F : Cav_B_Charlie_base_F { //General FTL : 27kg
...
+
+    //binoculars[] = {""};
+
+    magazines[] = {
+        "rhs_mag_30Rnd_556x45_M855A1_PMAG",15,
+        "rhs_mag_M433_HEDP",6,
+        "rhs_mag_M714_White",4,
+        "rhs_mag_m67",2,
+        "SmokeShell",4,
+        "SmokeShellGreen",
```
Comment (ColdEvul) [inline cScripts/Loadouts/CfgLoadouts_Charlie.hpp]: Remove last comma

File: cScripts/Loadouts/CfgLoadouts_Common.hpp [hunk 8]
```diff
@@ -200,30 +200,33 @@ class Cav_B_Bravo_Logi_base_F : Cav_B_Bravo_base_F {
...
     binoculars[] = {""};
 
     magazines[] = {
-        "rhs_mag_30Rnd_556x45_M855A1_Stanag",8,
-        "rhs_mag_m67",4,
-        "SmokeShell",6
+        "rhs_mag_30Rnd_556x45_M855A1_PMAG",15,
+        "rhs_mag_m67",2,
+        "SmokeShell",4,
+        "SmokeShellGreen",
```
Comment (ColdEvul) [inline cScripts/Loadouts/CfgLoadouts_Common.hpp]: Remove last comma

File: cScripts/Loadouts/CfgLoadouts_Charlie.hpp [hunk 9]
```diff
@@ -79,32 +152,40 @@ class Cav_B_C_RiflemanLAT_F : Cav_B_Charlie_base_F { //General Rifleman AT : 29k
...
+    //binoculars[] = {""};
 
     magazines[] = {
-        "rhs_mag_30Rnd_556x45_M855A1_Stanag",8,
-        "rhs_mag_m67",4,
-        "SmokeShell",6
+        "rhsusf_200rnd_556x45_mixed_soft_pouch_coyote",7,
+        "rhs_mag_m67",2,
+        "SmokeShell",4,
+        "SmokeShellGreen",
```
Comment (ColdEvul) [inline cScripts/Loadouts/CfgLoadouts_Charlie.hpp]: Remove last comma

**Comment by ColdEvul** [comment]
I'll handle the merge conflict it's the updated Bravo and Alpha loadouts that is spooking :)

**Comment by ColdEvul** [comment]
Snapshot `f9d8747` have been uploaded

**Comment by Robot-Panda15** [comment]
Brostrom can we get a new mission file up asap with these fixes

**Comment by ColdEvul** [comment]
yes sure

**Comment by ColdEvul** [comment]
Snapshot `c5ea9ae` have been uploaded

**Comment by ColdEvul** [comment]
Snapshot created for `fa85956`:
[cScripts-DEVBUILD-fa85956_TestBuild.zip](https://github.com/7Cav/cScripts/files/3654355/cScripts-DEVBUILD-fa85956_TestBuild.zip)

**Comment by ColdEvul** [comment]
Snapshot created for `37cc7b2`:
[cScripts-DEVBUILD-37cc7b2_TestBuild.zip](https://github.com/7Cav/cScripts/files/3665272/cScripts-DEVBUILD-37cc7b2_TestBuild.zip)

**Pull request merged**